#!/usr/bin/env python

PACKAGE='srsnode_odometry'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

####################################################################################################
gen.add("motor_count_per_rev", 
        int_t, 0,
        "Counts per motor revolution" , 
        4096, 0, 10000
)

gen.add("gearbox_ratio", 
        double_t, 0,
        "Gearbox reduction ratio" , 
        10.0, 0.0, 30.0
)

gen.add("robot_wheelbase_length", 
        double_t, 0,
        "Wheelbase length between two wheel contact points in meters" , 
        0.5235, 0.0, 1.0
)

gen.add("robot_leftwheel_radius", 
        double_t, 0,
        "Left wheel radius in meters" , 
        0.10243, 0.0, 0.3
)

gen.add("robot_rightwheel_radius", 
        double_t, 0,
        "Right wheel radius in meters" , 
        0.10243, 0.0, 0.3
)

####################################################################################################
exit(gen.generate(PACKAGE, PACKAGE, "RobotSetup"))